The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/13354554596/job/37295465205) [(plain)](https://github.com/rust-lang/rust/commit/ba4bcc12b9b10fb9992e68fc122ba21be3af8c7d/checks/37295465205/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
info: removing rustup binaries
info: rustup is uninstalled
##[group]Image checksum input
mingw-check-tidy
# We use the ghcr base image because ghcr doesn't have a rate limit
# and the mingw-check-tidy job doesn't cache docker images in CI.
FROM ghcr.io/rust-lang/ubuntu:22.04
ARG DEBIAN_FRONTEND=noninteractive
RUN apt-get update && apt-get install -y --no-install-recommends \
  g++ \
  make \
---

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#12 4.172   Downloading filelock-3.17.0-py3-none-any.whl (16 kB)
#12 4.189 Collecting distlib<1,>=0.3.7
#12 4.191   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#12 4.198      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 98.1 MB/s eta 0:00:00
#12 4.230 Collecting platformdirs<5,>=3.9.1
#12 4.233   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#12 4.315 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#12 4.496 Successfully installed distlib-0.3.9 filelock-3.17.0 platformdirs-4.3.6 virtualenv-20.29.2
#12 DONE 4.6s

#13 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#13 DONE 0.0s
---
DirectMap4k:      124864 kB
DirectMap2M:     6166528 kB
DirectMap1G:    12582912 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py,cpp
    Finished `dev` profile [unoptimized] target(s) in 0.05s
##[endgroup]
WARN: currently no CI rustc builds have rustc debug assertions enabled. Please either set `rust.debug-assertions` to `false` if you want to use download CI rustc or set `rust.download-rustc` to `false`.
downloading https://static.rust-lang.org/dist/2025-02-08/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
---
fmt check
fmt: checked 5845 files
tidy check
tidy: Skipping binary file check, read-only filesystem
##[error]tidy error: /checkout/src/ci/docker/scripts/musl.sh:49: tab character
##[error]tidy error: /checkout/src/ci/docker/scripts/musl.sh:50: tab character
##[error]tidy error: /checkout/src/ci/docker/scripts/musl.sh:51: tab character
##[error]tidy error: /checkout/src/ci/docker/scripts/musl.sh:52: tab character
##[error]tidy error: /checkout/src/ci/docker/scripts/musl.sh:53: tab character
##[error]tidy error: /checkout/src/ci/docker/scripts/musl.sh:54: tab character
##[error]tidy error: /checkout/src/ci/docker/scripts/musl.sh:55: tab character
##[error]tidy error: /checkout/src/ci/docker/scripts/musl.sh:56: tab character
##[error]tidy error: /checkout/src/ci/docker/scripts/musl.sh:61: line longer than 100 chars
##[error]tidy error: /checkout/src/ci/docker/scripts/musl.sh:62: tab character
##[error]tidy error: /checkout/src/ci/docker/scripts/musl.sh:63: tab character
##[error]tidy error: /checkout/src/ci/docker/scripts/musl.sh:64: tab character
##[error]tidy error: /checkout/src/ci/docker/scripts/musl.sh:65: tab character
##[error]tidy error: /checkout/src/ci/docker/scripts/musl.sh:66: tab character
##[error]tidy error: /checkout/src/ci/docker/scripts/musl.sh:67: tab character
##[error]tidy error: /checkout/src/ci/docker/scripts/musl.sh:68: tab character
##[error]tidy error: /checkout/src/ci/docker/scripts/musl.sh:69: tab character
##[error]tidy error: /checkout/src/ci/docker/scripts/musl.sh:70: tab character
##[error]tidy error: /checkout/src/ci/docker/scripts/musl.sh:71: tab character
removing old virtual environment
creating virtual environment at '/checkout/obj/build/venv' using 'python3.10' and 'venv'
creating virtual environment at '/checkout/obj/build/venv' using 'python3.10' and 'virtualenv'
Requirement already satisfied: pip in ./build/venv/lib/python3.10/site-packages (25.0.1)
All checks passed!
checking python file formatting
28 files already formatted
checking C++ file formatting
checking C++ file formatting
some tidy checks failed
Command has failed. Rerun with -v to see more details.
  local time: Sun Feb 16 11:33:35 UTC 2025
  network time: Sun, 16 Feb 2025 11:33:35 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/src/ci/docker/scripts/musl.sh
+++ b/src/ci/docker/scripts/musl.sh
@@ -30,6 +30,47 @@ MUSL=musl-1.2.3
 # may have been downloaded in a previous run
 if [ ! -d $MUSL ]; then
   curl https://www.musl-libc.org/releases/$MUSL.tar.gz | tar xzf -
+
+  # Apply patches for CVE-2025-26519. At the time of adding these patches no release containing them
+  # has been published by the musl project, so we just apply them directly on top of the version we
+  # were distributing already. The patches should be removed once we upgrade to musl >= 1.2.6.
+  #
+  # Advisory: https://www.openwall.com/lists/musl/2025/02/13/1
+  #
+  # Patches applied:
+  # - https://www.openwall.com/lists/musl/2025/02/13/1/1
+  # - https://www.openwall.com/lists/musl/2025/02/13/1/2
+  #
+  # ignore-tidy-tab
+  # ignore-tidy-linelength
+  patch -p1 -d $MUSL <<EOF
+--- a/src/locale/iconv.c
++++ b/src/locale/iconv.c
+@@ -502,7 +502,7 @@ size_t iconv(iconv_t cd, char **restrict in, size_t *restrict inb, char **restri
+ 			if (c >= 93 || d >= 94) {
+ 				c += (0xa1-0x81);
+ 				d += 0xa1;
+-				if (c >= 93 || c>=0xc6-0x81 && d>0x52)
++				if (c > 0xc6-0x81 || c==0xc6-0x81 && d>0x52)
+ 					goto ilseq;
+ 				if (d-'A'<26) d = d-'A';
+ 				else if (d-'a'<26) d = d-'a'+26;
+EOF
+  patch -p1 -d $MUSL <<EOF
+--- a/src/locale/iconv.c
++++ b/src/locale/iconv.c
+@@ -545,6 +545,10 @@ size_t iconv(iconv_t cd, char **restrict in, size_t *restrict inb, char **restri
+ 				if (*outb < k) goto toobig;
+ 				memcpy(*out, tmp, k);
+ 			} else k = wctomb_utf8(*out, c);
++			/* This failure condition should be unreachable, but
++			 * is included to prevent decoder bugs from translating
++			 * into advancement outside the output buffer range. */
++			if (k>4) goto ilseq;
+ 			*out += k;
+ 			*outb -= k;
+ 			break;
+EOF
 fi
 
 cd $MUSL

